semantic-core/src/Analysis/Concrete.hs

```diff
@@ -41,7 +41,7 @@ newtype FrameId = FrameId { unFrameId :: Precise }
   deriving (Eq, Ord, Show)
 
 data Concrete term
-  = Closure Loc Name term Env
+  = Closure Path Span Name term Env
```

Comment:
The original goal of annotating terms with `Loc`s was to ensure that when we call a function and jump to its body, we can set the correct path so that debugging/file directives reflect the location at the body of the closure, not the call site.

That goal is easily satisfied by simply keeping the path on the closure and setting it locally while evaluating the body, which we were actually already doing.